lib/internal/test_runner/harness.js

```diff
@@ -165,6 +167,15 @@ function setup(root) {
       // Run global before/after hooks in case there are no tests
       await root.run();
     }
+
+    if (root.subtestsPromise) {
+      // Wait for all subtests to finish, but keep the process alive in case
+      // there are no ref'ed handles left.
+      const keepAlive = setInterval(() => {}, TIMEOUT_MAX);
```

Comment:
Would this keep the runner running indefinitely?

Reply:
Yes, until any subtests finish running, and then the interval is cleared. Otherwise, the process can exit if there are no ref'ed handles.